runtime/vam/expr/cast/string.go

```diff
@@ -115,7 +115,7 @@ func castToString(vec vector.Any, index []uint32) (vector.Any, []uint32, bool) {
 			offs = append(offs, uint32(len(bytes)))
 		}
 	}
-	return vector.NewString(offs, bytes, nulls), nil, true
+	return vector.NewString(vector.NewBytesTable(offs, bytes), nulls), nil, true
```

Comment:
Nit: This happens often enough that I think it's worth continuing to have NewBytes accept slices and adding a new constructor like NewBytesFromTable (or else having NewBytes accept a table and adding NewBytesFromSlices, but I think the slice version will get called a lot more than the table version). Same goes for NewString.